efi/preinstall/profile.go

```diff
@@ -65,26 +72,32 @@ func (s authorityTrustDataSet) determineTrust(certs []*X509CertificateID) author
 			if auth.Authority.SignatureAlgorithm != cert.SignatureAlgorithm() {
 				continue
 			}
-			certTrust = auth.Trust
+
+			if flags&auth.Trust != flags {
+				return authoritiesNotTrusted
+			}
+			certFound = true
 			break
 		}
-		trust &= certTrust
+		if !certFound {
+			return authoritiesTrustUnknown
```

Comment:
maybe this should have a comment that the authority of this cert is not in the data set